crypto unit tests coverage

yasiramin10p opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/docker/docker/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

mixja committed:
Add awslogs multiline support

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mixja committed:
Add awslogs benchmarks

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mixja committed:
Formatting

Signed-off-by: Justin Menga <justin.menga@gmail.com>

Commit:
awslogs: use github.com/pkg/errors to wrap an error

Signed-off-by: Tibor Vass <tibor@docker.com>

dongluochen committed:
support cluster events

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

aboch committed:
Support configuration networks

- They are configuration-only networks which
  can be used to supply the configuration
  when creating regular networks.
- They do not get allocated and do net get plumbed.
  Drivers do not get to know about them.
- They can be removed, once no other network is
  using them.
- When user creates a network specifying a
  configuration network for the config, no
  other network specific configuration field
  is are accepted. User can only specify
  network operator fields (attachable, internal,...)

Signed-off-by: Alessandro Boch <aboch@docker.com>

aboch committed:
Allow to control network scope

- User can now promote a network's scope to swarm,
  so that the network is visible to swarm

Signed-off-by: Alessandro Boch <aboch@docker.com>

aboch committed:
Integrate local datascope network with swarm

Signed-off-by: Alessandro Boch <aboch@docker.com>

moypray committed:
fix when rpc reports "transport is closing" error, health check go routine will exit

Signed-off-by: Wentao Zhang <zhangwentao234@huawei.com>

tophj-ibm committed:
[pkg/term] switch syscall to x/sys

Switches calls to syscall to x/sys, which is more up to date.
This is fixes a number of possible bugs on other architectures
where ioctl tcget and tcset aren't implemented correctly.

There are a few remaining syscall references, because x/sys doesn't
have an Errno implementation yet.

Also removes a ppc64le and cgo build tag that fixes building on
ppc64le without cgo

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

dnephin committed:
Refactor builder.calcCopyInfo and dependencies

Remove pathCache from imageContexts
Extract validateCopySourcePath
Extract copyWithWildcards
Extract copyInfoForFile and walkSource from calcCopyInfo
Move copy internals to copy.go
remove source from Builder

Signed-off-by: Daniel Nephin <dnephin@docker.com>

Commit:
fix inconsistent state string with containerd

should be `stopped` according to containerd:
  https://github.com/containerd/containerd/blob/v0.2.x/runtime/runtime.go#L104

Signed-off-by: Deng Guangxing <dengguangxing@huawei.com>

chrisdias committed:
Support FROM...AS syntax for multi stage build dockerfiles

Signed-off-by: chrisdias <cdias@microsoft.com>

mlaventure committed:
Use containerd Status variable when checking container state

Signed-off-by: Kenfe-Mickael Laventure <mickael.laventure@gmail.com>

utzb committed:
adding rpm builds for ppc64le and s390x

Signed-off-by: Christy Perez <christy@linux.vnet.ibm.com>
Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>
Signed-off-by: Dominik Dingel <dingel@de.ibm.com>
Signed-off-by: Thorsten Winkler <thorsten.winkler@de.ibm.com>
Signed-off-by: Utz Bacher <utz.bacher@de.ibm.com>

abhi committed:
Adding network specific options to service create/update

The commit adds capability to accept csv parameters
for network option in service create/update commands.The change
includes name,alias driver options specific to the network.
With this the following will be supported

docker service create --name web --network name=docknet,alias=web1,driver-opt=field1=value1 nginx
docker service create --name web --network docknet nginx
docker service update web --network-add name=docknet,alias=web1,driver-opt=field1=value1
docker service update web --network-rm docknet

Signed-off-by: Abhinandan Prativadi <abhi@docker.com>

darstahl committed:
Revendor runtime-spec

Signed-off-by: Darren Stahl <darst@microsoft.com>

darstahl committed:
Use CpuMaximum instead of CpuPercent for more precision

Signed-off-by: Darren Stahl <darst@microsoft.com>

mixja committed:
Code review changes

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mixja committed:
Append new line to multiline events

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mixja committed:
Add missing month in datetime map

Signed-off-by: Justin Menga <justin.menga@gmail.com>

aboch committed:
Vendoring libnetwork @b2bc1a6

Signed-off-by: Alessandro Boch <aboch@docker.com>

mixja committed:
Code review changes

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mixja committed:
Update comments describing new behaviour

Signed-off-by: Justin Menga <justin.menga@gmail.com>

mlaventure committed:
Add missing go-difflib vendoring

Signed-off-by: Kenfe-Mickael Laventure <mickael.laventure@gmail.com>

tonistiigi committed:
Add builder dev-report for 2017-05-15

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

thaJeztah committed:
Moby report 2017-05-15

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

abhi committed:
swarmkit vendoring

Signed-off-by: Abhinandan Prativadi <abhi@docker.com>

Commit:
[pkg/term] Refactor BSD-specific files

Signed-off-by: Tibor Vass <tibor@docker.com>

tophj-ibm committed:
Rename term to winsize, tc_other to tc

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

Commit:
make.ps1 default to just building daemon

Signed-off-by: John Howard <jhoward@microsoft.com>

aboch committed:
Vendor swarmkit @998a47f

Signed-off-by: Alessandro Boch <aboch@docker.com>

nishanttotla committed:
Adding compatible platforms to service spec

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

aboch committed:
Vendor gogoprotobuf v0.4

Signed-off-by: Alessandro Boch <aboch@docker.com>

cpuguy83 committed:
Merge pull request #30891 from mixja/awslogs-multiline-support

Add awslogs multiline support

cpuguy83 committed:
Merge pull request #33075 from x1022as/stop

fix inconsistent state string with containerd

cyphar committed:
apparmor: make pkg/aaparser work on read-only root

This is necessary because normally `apparmor_parser -r` will try to
create a temporary directory on the host (which is not allowed if the
host has a rootfs). However, the -K option bypasses saving things to the
cache (which avoids this issue).

  % apparmor_parser -r /tmp/docker-profile
  mkstemp: Read-only file system
  % apparmor_parser -Kr /tmp/docker-profile
  %

In addition, add extra information to the ensureDefaultAppArmorProfile
errors so that problems like this are easier to debug.

Fixes: 2f7596aaef3a ("apparmor: do not save profile to /etc/apparmor.d")
Signed-off-by: Aleksa Sarai <asarai@suse.de>

Commit:
Merge pull request #33235 from Microsoft/jjh/tweakmakeps1

make.ps1 default to just building daemon

tiborvass committed:
Merge pull request #33180 from tophj-ibm/switch-pkg-term-syscalls-to-x/sys

[pkg/term] refactor and switch syscall to x/sys

Commit:
client: Expose DaemonHost to client users

This commit exposes `Client.host` as `Client.DaemonHost()`
This allows users of the client, a CLI for example, to query the Host
that the client is attempting to contact and vary their behaviour
accordingly. For example, to allow client-side configuration of
HTTP proxy settings for a number of different docker hosts.

Signed-off-by: Dave Tucker <dt@docker.com>

aboch committed:
Docs and manual changes

- for service create on node-local networks

Signed-off-by: Alessandro Boch <aboch@docker.com>

tiborvass committed:
Merge pull request #33233 from tiborvass/report-2017-05-15

Add dev report for may 15th

Commit:
Merge pull request #32421 from dongluochen/swarmkit_events

support cluster events

nishanttotla committed:
Updating test for compatible platforms to test unmarshal body

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

tonistiigi committed:
Merge pull request #32986 from moypray/containerd_close

fix when rpc reports "transport is closing" error, health check go routine will exit

Commit:
api: Only return a Platform when relevant information is available

I noticed that we could return a Platform that has no information filled
in. This doesn't look like it would cause any problems, but it would be
confusing. Fix the handler to only append to this slice when the
Platform is not empty.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

mavenugo committed:
Merge pull request #32981 from aboch/nlo2

Support swarm-mode services with node-local networks

Commit:
Fix checkpoint ls

Use create=false when calling getCheckpointDir
Fix #33263

Signed-off-by: Jacob Wen <jian.w.wen@oracle.com>

mavenugo committed:
Merge pull request #33239 from nishanttotla/add-platform-information-service-spec

Adding compatible platforms to service spec

mavenugo committed:
Merge pull request #33130 from abhinandanpb/network-opt

 Adding network specific options to service create/update

Commit:
ineffectual assignment to isCanonical, delete it

Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

Commit:
Merge pull request #33116 from dnephin/refactor-builder-copy

[Builder] Refactor COPY/ADD dispatchers

tophj-ibm committed:
[pkg/term] switch more syscalls to /x/sys/

Switches the remaining syscalls except Errno to /x/sys/.
This was supposed to be part of 33180

Signed-off-by: Christopher Jones <tophj@linux.vnet.ibm.com>

masahide committed:
Improved poor memory efficiency of awslogs

Signed-off-by: YAMASAKI Masahide <masahide.y@gmail.com>

Commit:
Update CLI commit hash

Since this new version of the CLI resolves image digests for swarm
services by default, and we do not want integration tests to talk to
Docker Hub, update CLI tests to suppress this behavior.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

thaJeztah committed:
Merge pull request #33271 from masahide/fix-append-loop-awslogs

Improved poor memory efficiency of awslogs

mlaventure committed:
Merge pull request #33264 from wenjianhn/chkp-ls

Fix checkpoint ls

thaJeztah committed:
Merge pull request #33250 from cyphar/apparmor-readonly-root-fix

apparmor: make pkg/aaparser work on read-only root

thaJeztah committed:
Merge pull request #33267 from YuPengZTE/devImageWithDigestString

ineffectual assignment to isCanonical, delete it, and make the "if" sentence to fit the golang usage

tonistiigi committed:
Use chrootarchive for plugin rootfs

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

nishanttotla committed:
Ensure service images get default tag and print familiar strings

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

mlaventure committed:
Merge pull request #33274 from aaronlehmann/update-cli

Update CLI commit hash

mavenugo committed:
Vendoring libnetwork

Signed-off-by: Madhu Venugopal <madhu@docker.com>

dmcgowan committed:
Support overlay2 on btrfs

OverlayFS is supported on top of btrfs as of Linux Kernel 4.7.
Skip the hard enforcement when on kernel 4.7 or newer and
respect the kernel check override flag on older kernels.
https://btrfs.wiki.kernel.org/index.php/Changelog#By_feature

Signed-off-by: Derek McGowan <derek@mcgstyle.net>

nishanttotla committed:
Adding unit tests for pin by digest (client)

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

thaJeztah committed:
Merge pull request #33269 from tophj-ibm/switch-more-syscalls-pkg-term

[pkg/term] switch more syscalls to /x/sys/

thaJeztah committed:
Merge pull request #33278 from tonistiigi/plugin-chroot

Use chrootarchive for plugin rootfs

thaJeztah committed:
Merge pull request #33281 from dmcgowan/overlay2-on-btrfs

Support overlay2 on btrfs

aboch committed:
Properly identify ingress network created with older swarm

- otherwise docker network prune will remove it

Signed-off-by: Alessandro Boch <aboch@docker.com>

cpuguy83 committed:
Merge pull request #33280 from mavenugo/vin-ln-1706

Vendoring libnetwork b2bc1a684

Commit:
arg filt

Signed-off-by: yupengzte <yu.peng36@zte.com.cn>

fd committed:
Remove deadlock in ContainerWait

when `cli.post(...)` fails `errC <- err` blocks because `errC` is unbufferd.

Signed-off-by: Simon Menke <simon.menke@gmail.com>

Commit:
crypto unit tests coverage

vdemeester committed:
Remove more opts unused in docker/docker

These are already present in `docker/cli` 👼

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

dongluochen committed:
Event tests need to wait for events

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

thaJeztah committed:
Improve description of Running and Paused booleans

Commit abd72d4008dde7ee8249170d49eb4bc963c51e24 added
a "FIXME" comment to the container "State", mentioning
that a container cannot be both "Running" and "Paused".

This comment was incorrect, because containers on
Linux actually _must_ be running in order to be
paused.

This patch adds additional information both in a
comment, and in the API documentation to clarify
that these booleans are not mutually exclusive.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah committed:
Merge pull request #33291 from YuPengZTE/devFilter

arg filter for printf verb %s of wrong type

cpuguy83 committed:
Merge pull request #33296 from vdemeester/more-opts-cleanup

Remove more opts unused in docker/docker

cpuguy83 committed:
Merge pull request #33286 from aboch/ingr

Properly identify ingress network created with older swarm

thaJeztah committed:
Merge pull request #33293 from fd/patch-1

Remove deadlock in ContainerWait

thaJeztah committed:
Merge pull request #33299 from thaJeztah/explain-relation-between-paused-and-stopped

Improve description of Running and Paused booleans

thaJeztah committed:
Merge pull request #33261 from aaronlehmann/empty-platform

api: Only return a Platform when relevant information is available

ehazlett committed:
remove RuntimeData from cluster and types

Signed-off-by: Evan Hazlett <ejhazlett@gmail.com>

darstahl committed:
Revendor go-winio to v0.4.1

Signed-off-by: Darren Stahl <darst@microsoft.com>

thaJeztah committed:
Merge pull request #33308 from ehazlett/remove-runtime-data

Remove RuntimeData from cluster and types

Commit:
Add a restarting check to ContainerAttach

Signed-off-by: yangshukui <yangshukui@huawei.com>

thaJeztah committed:
Merge pull request #33307 from dongluochen/swarm_event_test

Event tests need retry logic

thaJeztah committed:
Merge pull request #33253 from dave-tucker/clientHost

client: Expose ClientHost to client users

mlaventure committed:
Merge pull request #32774 from darrenstahlmsft/ProcessorMaximum

Use CpuMaximum instead of CpuPercent for more precision

Commit:
Merge pull request #33310 from darrenstahlmsft/RevendorGowinio

Revendor go-winio to v0.4.1

thaJeztah committed:
Merge pull request #33279 from nishanttotla/fix-service-image-issues

Ensure service images get default tag and print familiar strings

cyli committed:
Do not clear swarm directory at the begining of swarm init and swarm join now.
However, do clear the directory if init or join fails, because we don't
want to leave it in a half-finished state.

Signed-off-by: Ying Li <ying.li@docker.com>

Commit:
vendor: Remove docker-credential-helpers

This appears to be a remnant from the CLI that is no longer imported
anywhere. Remove it from vendor.conf and the vendor directory.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

cpuguy83 committed:
Merge pull request #33342 from aaronlehmann/cred-helpers-vendor

vendor: Remove docker-credential-helpers

Commit:
Merge pull request #33341 from cyli/do-not-clear-state-on-swarm-init-join

Do not clear swarm directory on `swarm init` and `swarm join`

vdemeester committed:
Fix `request.SockRequestRaw` error check

We should check for error before reading the response (response can be
nil, and thus this would panic)

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

darstahl committed:
Update go-winio to v0.4.2

Signed-off-by: Darren Stahl <darst@microsoft.com>

Commit:
Covered Crypto test for registry/endpoint_test.go and registry/registry_test.go

usmanniqbal commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
 	}
 }
 
+func TestNewV1EndPointFromStr(t *testing.T){
```
Is this test really required?

usmanniqbal commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
+	require.EqualValues(t,e.URL.Host,"index.docker.io")
+}
+
+func TestNewV1Endpoint(t *testing.T)  {
```
Same as above, and seems that its only constructing the struct.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
 }
 
+func TestNewV1EndPointFromStr(t *testing.T){
+	e, err := newV1EndpointFromStr("https://index.docker.io/v1/", nil, "", nil)
```
Use proper naming for variables, both (e, err) sound like error.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
 
+func TestNewV1EndPointFromStr(t *testing.T){
+	e, err := newV1EndpointFromStr("https://index.docker.io/v1/", nil, "", nil)
+	require.NotNil(t, e)
```
This function `newV1EndpointFromStr` will get tested in `TestEndpointParse` test, so no need to write its test unless you are passing it configurations different from what already used in `TestEndpointParse`, which i don't think is the case.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
+func TestNewV1EndPointFromStr(t *testing.T){
+	e, err := newV1EndpointFromStr("https://index.docker.io/v1/", nil, "", nil)
+	require.NotNil(t, e)
+	require.Nil(t, err)
```
Use assert.NoError(t, err) for checking errors.

usmanniqbal commented on registry/registry_test.go:
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
+	require.EqualValues(t,tlsConfigDefault.ServerName,resp.ServerName)
+}
+
+func TestReadCertsDirectory(t *testing.T)  {
```
I dont think so we need to write tests for reading files only, and its redundant too as newTLSConfig also using it internally.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
+
+func TestNewV1Endpoint(t *testing.T)  {
+	uri, err := url.Parse("http://0.0.0.0:5000/v2/")
+	require.Nil(t,err)
```
Use assert.NoError(t, err) for checking errors.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
+	var index = makeIndex("/v1/")
+	tlsConfig, err := newTLSConfig(index.Name, index.Secure)
+	require.Nil(t,err)
+	e, err := newV1Endpoint(*uri,tlsConfig,"",nil)
```
same as mentioned above, variable renaming issue.

uzairalikhan commented on registry/endpoint_test.go:
```diff
@@ -46,6 +47,25 @@ func TestEndpointParseInvalid(t *testing.T) {
+	require.Nil(t,err)
+	var index = makeIndex("/v1/")
+	tlsConfig, err := newTLSConfig(index.Name, index.Secure)
+	require.Nil(t,err)
```
same as mentioned above for checking errors.

uzairalikhan commented on registry/registry_test.go:
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
+
+func TestNewTransport(t *testing.T) {
+	var index = makeIndex("/v1/")
+	resp, err := newTLSConfig(index.Name,index.Secure)
```
Use tlsConfig rather than resp.

uzairalikhan commented on registry/registry_test.go:
```diff
@@ -893,6 +899,34 @@ func TestIsSecureIndex(t *testing.T) {
 
+
+func TestTLSConfig(t *testing.T){
+	var index = makeIndex("/v1/")
```
Index variable is used many times and always re declared, its better to  initialize it at the top then use it in all the tests.

uzairalikhan commented on registry/registry_test.go:
```diff
@@ -14,6 +14,12 @@ import (
+	"github.com/docker/go-connections/tlsconfig"
+
+	"path/filepath"
+	"os"
```
Use proper ordering for imports and always run `gofmt` on your code before making any PR.

nhorman committed:
Ensure that a device mapper task is referenced until task is complete

DeviceMapper tasks in go use SetFinalizer to clean up C construct
counterparts in the C LVM library.  While thats well and good, it relies
heavily on the exact interpretation of when the golang garbage collector
determines that an object is unreachable is subject to reclaimation.
While common sense would assert that for stack variables (which these DM
tasks always are), are unreachable when the stack frame in which they
are declared returns, thats not the case.  According to this:

https://golang.org/pkg/runtime/#SetFinalizer

The garbage collector decides that, if a function calls into a
systemcall (which task.run() always will in LVM), and there are no
subsequent references to the task variable within that stack frame, then
it can be reclaimed.  Those conditions are met in several devmapper.go
routines, and if the garbage collector runs in the middle of a
deviceMapper operation, then the task can be destroyed while the
operation is in progress, leading to crashes, failed operations and
other unpredictable behavior.

The fix is to use the KeepAlive interface:

https://golang.org/pkg/runtime/#KeepAlive

The KeepAlive method is effectively an empy reference that fools the
garbage collector into thinking that a variable is still reachable.  By
adding a call to KeepAlive in the task.run() method, we can ensure that
the garbage collector won't reclaim a task object until its execution
within the deviceMapper C library is complete.

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>

thaJeztah committed:
Merge pull request #33167 from utzb/rpmbuilds

adding rpm builds for ppc64le and s390x

Commit:
Merge pull request #33095 from chrisdias/chrisdias/multistagebuild

syntax highlighting: support FROM...AS for multi stage build

tonistiigi committed:
Merge pull request #33358 from vdemeester/fix-integration-request-package

Fix `request.SockRequestRaw` error check

cpuguy83 committed:
Merge pull request #33376 from nhorman/keepalive

Ensure that a device mapper task is referenced until task is complete

thaJeztah committed:
Merge pull request #33365 from darrenstahlmsft/revendorGowinio

Update go-winio to v0.4.2

Commit:
Merge pull request #33292 from keloyang/attach-restart-check

Add a restarting check to ContainerAttach

Commit:
Changes fixed as per reviewer comments

Commit:
changes in file
